my_package/dataset/dataset.py

```diff
@@ -0,0 +1,317 @@
+from abc import ABC, abstractmethod
+
+
+class Data(Dataset):
+    def __init__(self, X_data, y_data, float=torch.float64):
+        self.X_data = X_data
+        self.y_data = y_data
+        self.float = float
+
+    def __getitem__(self, index):
+        return self.X_data[index].to(self.float), self.y_data[index].to(self.float)
+
+    def __len__(self):
+        return len(self.X_data)
+    
+    def append(self, X, y):
+        """
+        append new instances to the data
+        """
+        self.X_data = torch.cat((self.X_data, X), 0)
+        self.y_data = torch.cat((self.y_data, y), 0)
+
+
+class Branin_Data(Data):
+    """
+    Implements the "Data" class for Branin data. 
+    
+        X_data: array(N, 2) in the domain [0; grid_size]; states (i.e., grid positions) 
+        y_data: array(N, 1); scores at each position
+        normalize_scores: bool; maps the scores to [0; 1]
+        grid_size: int; specifies the width and height of the Branin grid (grid_size x grid_size); used to normalize the state positions
+        device: pytorch device used for calculations
+
+
+    """
+    def __init__(self, X_data, y_data, normalize_scores=True, grid_size=100, float=torch.float64):
+        super().__init__(X_data, y_data, float=float)
+        self.normalize_scores = normalize_scores
+        self.grid_size = grid_size
+        self.stats = self.get_statistics(y_data)
+        
+
+    def get_statistics(self, y):
+        """
+        called each time the dataset is updated so has the most recent metrics
+        """
+        dict = {}
+        dict["mean"] = torch.mean(y)
+        dict["std"] = torch.std(y)
+        dict["max"] = torch.max(y)
+        dict["min"] = torch.min(y)
+        return dict
+
+
+    def __getitem__(self, index):
+        X = self.X_data[index]
+        y = self.y_data[index]
+        return self.preprocess(X, y)
+    
+    
+    def append(self, X, y):
+        """
+        append new instances to the data
+        """
+        # X, y = self.deprocess(X, y) # append data in raw form (i.e., not normalized)
+        X /= self.grid_size
+        super().append(X, y)
+        self.stats = self.get_statistics(self.y_data) # update the score statistics
+    
+    def preprocess(self, X, y):
+        """
+        - normalizes the scores
+        - normalizes the states
+        """
+        states = X / self.grid_size
+        if self.normalize_scores:
+            scores = (y - self.stats["min"]) / (self.stats["max"] - self.stats["min"])
+
+        return states, scores
+    
+    # def deprocess(self, X, y):
+    #     """
+    #     - denormalizes the scores
+    #     - denormalizes the states
+    #     """
+    #     states = X * self.grid_size
+    #     # TODO: check if we have to denormalize the scores
+    #     if self.normalize_scores:
+    #         scores = y * (self.stats["max"] - self.stats["min"]) + self.stats["min"]
+
+    #     return states, scores
+
+    
+
+
+
+class AL_DatasetHandler(ABC):
```

Comment:
The (Python PEP) convention for class names is to use CapWords, that is separate words by the use of capital letters instead of underscores: https://peps.python.org/pep-0008/#class-names

So this would be `AlDatasetHandler` or `ALDatasetHandler`.